### ISSUE TITLE
Crashing in i phone when loading more the 1000 data and slowing down in desktop site.

### ISSUE THREAD

**Issue by Navin-vis** [vitalii-andriiovskyi/ngx-owl-carousel-o#58]
When api response more then 1000 data it make the carousel slow and crashing the site in i phone not able to handle the bulk of data in same time.

Thanks.

**Comment by vitalii-andriiovskyi** [comment]
Hi @Navin-vis 

I think you should limit data to show in the carousel. Maybe 100 items/slides would be enough. 
I don't know your implementation of the carousel. But if you have 1000 slides, the user won't be willing to see all of them.

**Comment by Navin-vis** [comment]
Yes but 100 is more for the carousel i currently using 10 data only and working  fine in i phone but i want more data to be fit in.
And also owl is crashing when zooming the site in i phone. 
I am dropping the video so you can have the idea what i am trying to do.
[videob756d47d-3402-4af0-a189-6da0a0e1bfa6video.mp4.zip](https://github.com/vitalii-andriiovskyi/ngx-owl-carousel-o/files/3979324/videob756d47d-3402-4af0-a189-6da0a0e1bfa6video.mp4.zip)

**Comment by vitalii-andriiovskyi** [comment]
The URL of your web-app is visible in the video. I checked the website and saw interesting moment: you don't use `ngx-owl-carousel-o` in your project. 

Probably you're using [`ngx-owl-carousel`](https://github.com/mujtaba01/ngx-owl-carousel). This library uses jQuery. [My library (`ngx-owl-carousel-o` )](https://github.com/vitalii-andriiovskyi/ngx-owl-carousel-o) has the same functionality, but it doesn't use jQuery, relies on Angular and has different usage in project.

**Comment by Navin-vis** [comment]
Ok, so you are saying this problem will not occur in this  "My library (ngx-owl-carousel-o )" library.

**Comment by vitalii-andriiovskyi** [comment]
Don't know.  Trying out it could answer.

**Comment by Navin-vis** [comment]
yup i tried it out, but zooming issue didn't  solve. i think it might be device issue.

**Comment by ruslanguns** [comment]
Is it any way possible to lazy load data with the carousel so there's no matter how many data comes from resource it just shows as more as it can ?

**Comment by vitalii-andriiovskyi** [comment]
@ruslanguns 

There're options `lazyLoad` and `lazyLoadEager`. But `lazyLoad` work just for content. 

For example, you have the array with length 100,  set `lazyLoad` to `true` and `items` to `3`. The carousel will have 100 slides and just 3 of them will have the content. 97 slides will be empty. If you navigate left/right the carousel will add the content to next/prev slide (4th/100th slide). 

If extra data will come from the server, let say data for 10 slides, the carousel will get rerendered and it will have 110 slides. But in this case it'll set the first slide as active. To set the slide which was viewed by the user before getting extra 10 slides, you have to reset the option `startPosition` assigning to it the index of the first slide from the last active slides.

**Comment by sunil512** [comment]
I set lazyLoad to true and I'm getting API response data more than 1000. It's not getting stuck/crushed but while clicking on prev or next slide it's taking 10sec of time to move to the next/prev slide. can it move next/prev slide immediately?
